syntax = "proto2";

package shardkvinner_messages;

message Data{
	required string key=1;
	required string value=2;
}

message Cache{
	required int64 key=1;
	required string value=2;
}

message ShardMigrationRequest{
	required int64 shard=1;
	required int64 confignum=2;
}

message ShardMigrationResponse{
	required int64 flag=1;
	required int64 shard=2;
	required int64 confignum=3;
	repeated Data data=4;
	repeated Cache cache=5;
}

message ShardCleanupRequest{
	required int64 shard=1;
	required int64 confignum=2;
}

message ShardCleanupResponse{
	required int64 flag=1;
	optional string leader_name=2;
}

service ShardkvinnerMessages{
	rpc ShardMigration (ShardMigrationRequest) returns (ShardMigrationResponse) {}
	rpc ShardCleanup (ShardCleanupRequest) returns (ShardCleanupResponse) {}
}
